### PR TITLE
Update samples to have more descriptive names

### PR DESCRIPTION
**Pull request by ijagberg**
Calling them "Function1", "Function2" etc. makes it quite hard to look
for examples of specific functionality.

### Issue describing the changes in this PR

resolves #issue_for_this_pr

### Pull request checklist

* [x] My changes **do not** require documentation changes
* [X] My changes **should not** be added to the release notes for the next release
* [X] My changes **do not** need to be backported to a previous version

### Additional information

Additional PR information

### CODE REVIEW

File: samples/FunctionApp/HttpTriggerWithFunctionContextLogger/HttpTriggerWithFunctionContextLogger.cs [hunk 1]
```diff
@@ -8,9 +8,9 @@
 
 namespace FunctionApp
 {
-    public static class Function4
+    public static class HttpTriggerWithFunctionContextLogger
     {
-        [Function("Function4")]
+        [Function(nameof(HttpTriggerWithFunctionContextLogger))]
```
Comment (kshyju) [inline samples/FunctionApp/HttpTriggerWithFunctionContextLogger/HttpTriggerWithFunctionContextLogger.cs:13]: I am more inclined to leave this as `HttpTriggerStatic` or `HttpTriggerSimple` ?

**Review by kshyju — APPROVED** [review]
Thank you @ijagberg 🎉

**Comment by ijagberg** [comment]
Cool, I agree with and have now implemented both your suggestions.

**Review by fabiocav — APPROVED** [review]

**Pull request merged**